Lib/test/libregrtest/save_env.py

```diff
@@ -292,6 +299,25 @@ def restore_warnings_showwarning(self, fxn):
         warnings = self.get_module('warnings')
         warnings.showwarning = fxn
 
+    def get_stty_echo(self):
+        termios = self.try_get_module('termios', demand=True)
+        if not os.isatty(fd := sys.__stdin__.fileno()):
+            return None
+        attrs = termios.tcgetattr(fd)
+        lflags = attrs[3]
```

Comment:
Are there other attributes that could be changed that we might want to restore?  ISTM it would be simpler to just save and restore the whole `attrs` list, but I don't know if there might be things that could reasonably change without being something that should be restored.

Reply:
I thought about this too, but decided that since `echo` is the only attribute that I've currently noticed affecting the terminal after the test run, I might as well keep it narrow for now.  We could expand on this in the future if needed, but I'd rather just fix the thing that is immediately noticeable for now.